### PR TITLE
feat: add pre-commit hooks

### PR DESCRIPTION
**Pull request by ishita1805**
**Issue** 
#197 

**Description** 
This PR adds the following format checkers as pre-commit hooks:
- eslint
- check-case-conflict
- pretty-format-json
- end-of-file-fixer
- mixed-line-ending
- trailing-whitespace
- double-quote-string-fixer

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+pre-commit
```
Comment (SagiPolaczek) [inline requirements_dev.txt:1]: Please move to `fuse/requirements_dev.txt`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: check-case-conflict
+      - id: pretty-format-json
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+      - id: double-quote-string-fixer
```
Comment (SagiPolaczek) [inline .pre-commit-config.yaml:10]: Note that this one contradicts `black` configuration, as it prefers double quotes upon single ones.

**Review by SagiPolaczek — CHANGES_REQUESTED** [review]
Thanks Ishita!

Note that the pre-commits hooks that we want to use are: `black`, `flake8` and `mypy`. Please consider visit [this](https://github.com/PyCQA/flake8/blob/main/.pre-commit-config.yaml) as a reference.
Also note that arguments should not be provided because we are using configuration files.

After applying those changes and the one in the in-line comment we shall merge.

Thanks and looking forward for your response @ishita1805 .

**Comment by ishita1805** [comment]
@SagiPolaczek I've updated the PR, please let me know if something else needs to be done

**Review by SagiPolaczek — APPROVED** [review]
Thanks Ishita! Looks good 😄

**Comment by SagiPolaczek** [comment]
@ishita1805 

Just to inform you:
We currently have some env issue with the CI/CD. Your PR has already been approved and will be merge as soon as we'll solve the issue.

Thanks a lot for your contribution!

**Review by mosheraboh — APPROVED** [review]
Thanks a lot @ishita1805! Looks good.
 @SagiPolaczek I'm merging, but please note that the versions in pre_commit of black and mypy are not aligned with the versions in requirements_dev.txt.
 I will merge it anyway and if necessary align them in another PR.

**Pull request merged**